MoltenVK/MoltenVK/GPUObjects/MVKImage.mm

```diff
@@ -1531,41 +1559,42 @@
 	}
 
     // Validate whether the image view configuration can be supported
-    if ( _image ) {
-        VkImageType imgType = _image->getImageType();
-        VkImageViewType viewType = pCreateInfo->viewType;
-
-        // VK_KHR_maintenance1 supports taking 2D image views of 3D slices for sampling.
-		// No dice in Metal. But we are able to fake out a 3D render attachment by making the Metal view
-		// itself a 3D texture (when we create it), and setting the rendering depthPlane appropriately.
-        if ((viewType == VK_IMAGE_VIEW_TYPE_2D || viewType == VK_IMAGE_VIEW_TYPE_2D_ARRAY) && (imgType == VK_IMAGE_TYPE_3D)) {
-            if (!mvkIsOnlyAnyFlagEnabled(_usage, VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT)) {
-                setConfigurationResult(reportError(VK_ERROR_FEATURE_NOT_PRESENT, "vkCreateImageView(): 2D views on 3D images can only be used as color attachments."));
-            }
-        }
+	if ( !_image->getIsValidViewFormat(pCreateInfo->format) ) {
+		setConfigurationResult(reportError(VK_ERROR_FORMAT_NOT_SUPPORTED, "vkCreateImageView(): The format is not in the list of allowed view formats declared in the VkImageFormatListCreateInfo provided in vkCreateImage()."));
+	}
```

Comment:
Isn't it [just undefined behavior](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/chap12.html#VUID-VkImageViewCreateInfo-pNext-01585) to use view formats outside that list?

Reply:
>Isn't it just undefined behavior to use view formats outside that list?

Hmmm...that rule has explicit normative terminology. I always read normative VU's to return an error, rather than undefined behavior. 

Admittedly, monitoring VU's is what the validation layer is doing, and drivers don't necessarily need to repeat that validation.

Creating a view is set-up code, so leaving a simple test like this in doesn't really cause a performance hit. But this does raise the issue that we are fairly arbitrary with our VU validations within MoltenVK. Sometimes we do them, sometimes not. 

Perhaps we should discuss a policy that says MoltenVK should only perform validation that is needed because of Metal limitations that cause non-conformance, but **_not_** perform validation that is defined by the spec, because the validation layers are responsible for that?

Reply:
> Perhaps we should discuss a policy that says MoltenVK should only perform validation that is needed because of Metal limitations that cause non-conformance, but **_not_** perform validation that is defined by the spec, because the validation layers are responsible for that?

That sounds reasonable. It's what I was thinking we should do.

Reply:
>That sounds reasonable. It's what I was thinking we should do.

Okay...let's adopt that from now on. I've removed the code above, and I've added text regarding a validation code policy to the _Contributing_ section of the `README.md` document. Let me know if that wording looks good.

Moving forward, we can each remove excess validation code by attrition as we come across it during other development.